crates/emitter/src/object_emitter.rs

```diff
@@ -0,0 +1,161 @@
+use crate::ast_emitter::AstEmitter;
+use crate::emitter::EmitError;
+use ast::source_atom_set::SourceAtomSetIndex;
+
+/// Struct for emitting bytecode for a property where its name is string.
+///
+/// The property name can be a string that contains index, and in that case
```

Comment:
maybe "If the property name is a string representing a number, it is considered an index. In this case, NamePropertyEmitter falls back internally to IndexPropertyEmitter"